### PR TITLE
add method to reset proxies when flag toggled

### PR DESCRIPTION
**Pull request by BelfordZ**
## Explanation

Fixes a bug where turning on / off the request queue feature flag requires dapps to be manually refreshed in order for the changes to take effect.

### `@metamask/selected-network-controller`

- **ADDED**: A method on selected-network-controller that can be called when changing the feature flag. This will update the proxies accordingly.
- **FIXED**: Fixed a bug where when the feature flag was off, we were not savng network selections for domains.

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/selected-network-controller/src/SelectedNetworkController.ts [hunk 1]
```diff
@@ -205,4 +204,14 @@ export class SelectedNetworkController extends BaseController<
...
   }
+
+  setPerDomainNetwork(enabled: boolean) {
+    this.update((state) => {
+      state.perDomainNetwork = enabled;
+      return state;
+    });
+    Object.entries(this.state.domains).forEach(([domain, networkClientId]) => {
+      this.setNetworkClientIdForDomain(domain, networkClientId);
+    });
```
Comment (adonesky1) [inline packages/selected-network-controller/src/SelectedNetworkController.ts:215]: Do we want to do this if `enabled` is false?

**Comment by adonesky1** [comment]
> FIXED: Fixed a bug where when the feature flag was off, we were not savng network selections for domains.

I think this was more of a change since I think this was intentional

File: packages/selected-network-controller/src/SelectedNetworkController.ts (hunk 1 above)
Comment (BelfordZ) [reply packages/selected-network-controller/src/SelectedNetworkController.ts:215]: ah yeah I missed that part! thank you! adding that now.

File: packages/selected-network-controller/tests/SelectedNetworkController.test.ts [hunk 2]
```diff
@@ -184,4 +184,25 @@ describe('SelectedNetworkController', () => {
...
+        setTarget: jest.fn(),
+        eventNames: jest.fn(),
+        rawListeners: jest.fn(),
+        removeAllListeners: jest.fn(),
+      };
+      createEventEmitterProxyMock.mockReturnValue(mockProviderProxy);
+      controller.setNetworkClientIdForDomain('example.com', 'network7');
+      expect(mockProviderProxy.setTarget).toHaveBeenCalledTimes(0);
+      controller.setPerDomainNetwork(true);
+      expect(mockProviderProxy.setTarget).toHaveBeenCalledTimes(2);
```
Comment (adonesky1) [inline packages/selected-network-controller/tests/SelectedNetworkController.test.ts:205]: [nit] could be good to check what these have been called with.

**Review by adonesky1 — APPROVED** [review]
LGTM! one non blocking request to update the test

**Pull request merged**